scm/etc/scripts/UFS_LSM_param.py

```diff
@@ -0,0 +1,432 @@
+#!/usr/bin/env python
+
+import argparse
+import os
+import fnmatch
+import logging
+from netCDF4 import Dataset
+import numpy as np
+from shapely.geometry import Point, Polygon
+import copy
+import math
+import re
+
+###############################################################################
+# Global settings                                                             #
+###############################################################################
+
+#Physical constants
+earth_radius = 6371000.0 #m
+
+# For developers: set logging level to DEBUG for additional output
+#LOGLEVEL = logging.DEBUG
+LOGLEVEL = logging.INFO
+
+###############################################################################
+# Command line arguments                                                      #
+###############################################################################
+
+parser = argparse.ArgumentParser()
+group1 = parser.add_mutually_exclusive_group(required=True)
+group1.add_argument('-l', '--location',   help='longitude and latitude in degress E and N, respectively, separated by a space', nargs=2, type=float)
+group1.add_argument('-ij','--index',      help='i,j indices within the tile (if known - bypasses search for closest model point to lon/lat location)', nargs=2, type=int)
+parser.add_argument('-g', '--grid_dir',   help='directory path containing FV3 tile supergrid files', required=True)
+parser.add_argument('-t', '--tile',       help='tile of desired point (if known - bypasses tile search if present)', type=int, choices=range(1,7))
+
+###############################################################################
+# Functions and subroutines                                                   #
+###############################################################################
+
+def parse_arguments():
+    """Parse command line arguments"""
+    args = parser.parse_args()
+    location = args.location
+    index = args.index
+    grid_dir = args.grid_dir
+    tile = args.tile
+    
+    #validate args
+    if not os.path.exists(grid_dir):
+        message = 'The directory {0} does not exist'.format(grid_dir)
+        logging.critical(message)
```

Comment:
Just for your info, it is possible to configure the logging handler such that a critical log entry automatically throws an Exception (see, for example https://stackoverflow.com/questions/48184168/python-logging-critical-to-raise-exception-and-dump-stacktrace-and-die). Not saying you should change anything, but if you generally want all Python scripts inside SCM to throw an Exception when a critical error is encountered, you could define a common module to use.

Reply:
Thanks, I'll look into this for the future.